src/remake.jl

```diff
@@ -58,6 +66,14 @@ function remake(prob::ODEProblem; f = missing,
 
     if p === missing
         p = prob.p
+    else
+        if p isa Dict || eltype(p) isa Pair
+            if hasproperty(prob.f, :sys) && hasfield(typeof(prob.f.sys), :ps)
+                p = handle_varmap(p, prob.f.sys, field = :ps, defaults = prob.f.sys.ps .=> prob.p)
```

Comment:
This parameters could depend on the initial condition, so you should include `prob.u0` map as well.